internal/command/jsonstate/state.go

```diff
@@ -561,8 +587,27 @@ func SensitiveAsBool(val cty.Value) cty.Value {
 		// marshal to JSON anyway, and all of these mapping types are
 		// indistinguishable in JSON.
 		return cty.ObjectVal(vals)
+	case ty.IsObjectType():
+		if !val.IsKnown() || ty.Equals(cty.EmptyObject) {
```

Comment:
Why is the obj check moved separately? As far as I understand, `IndexStep` supports objects as well as maps, and also objects could be iterated via `ElementIterator` so it could be the same code for both maps and objects. WDYT?

Reply:
`IndexStep` for map and `GetAttrStep` for object